library/components/TTooltip.vue

```diff
@@ -1,64 +1,122 @@
+		}
+	},
 	data: () => ({
-		tooltipPositions: {}, // This object should be empty by default to only apply related properties (top, left...)
-        arrowPositions: {
-            top: "left-0 right-0 -bottom-[7px]",
-            bottom: "left-0 right-0 -top-[7px]",
-            left: "top-2 -right-[7px]",
-            right: "top-2 -left-[7px]",
-        }
+		positions: {},
+		showTooltip: false,
+		isForcedOpen: false
 	}),
 	mounted() {
-		this.placeTooltip();
+		onClickOutside(this.$refs.tooltipContainer, () => {
+			this.showTooltip = false;
+		});
 	},
 	methods: {
 		placeTooltip() {
+			// Calculate and place both tooltip and arrow.
+			const triggerElement = this.$refs.trigger;
+			const tooltipElement = this.$refs.tooltip;
+			const tooltipPositions = {};
+			const positions = { arrow: {}, tooltip: {} };
+
+			if (triggerElement && tooltipElement) {
+				const position = triggerElement.getBoundingClientRect();
 
-            // Place tooltip near parent according to `position` prop.
-			const parent = this.$parent.$el;
-            const tooltipPositions = {}
-			if (parent) {
-				const position = parent.getBoundingClientRect();
 				if (this.position === "top") {
-					// Push from bottom
-					tooltipPositions.top = "-" + (position.bottom + 15) + "px";
+					positions.tooltip.top =
+						position.top - (tooltipElement.clientHeight + 10) + "px";
+					positions.tooltip.left =
+						position.left - tooltipElement.clientWidth / 2 + "px";
+					positions.arrow.top = position.top - 20 + "px";
+					positions.arrow.left = position.left + "px";
 				} else if (this.position === "bottom") {
-					// Push from top
-					tooltipPositions.top = (parent.clientHeight + 15) + "px";
+					positions.tooltip.top =
+						position.top + triggerElement.clientHeight + 15 + "px";
+					positions.tooltip.left =
+						position.left - tooltipElement.clientWidth / 2 + "px";
+					positions.arrow.top =
+						position.top + triggerElement.clientHeight + 10 + "px";
+					positions.arrow.left = position.left + "px";
 				} else if (this.position === "left") {
-                    // Place on left side
-                    tooltipPositions.right = (position.width + 15) + "px";
-                } else {
-                    // Place on right side
-                    tooltipPositions.left = (position.width + 15) + "px";
-                }
+					positions.tooltip.top =
+						position.top - tooltipElement.clientHeight / 2 + "px";
+					positions.tooltip.left =
+						position.left - tooltipElement.clientWidth - 15 + "px";
+					positions.arrow.top = position.top + "px";
+					positions.arrow.left = position.left - 25 + "px";
+				} else {
+					positions.tooltip.top =
+						position.top - tooltipElement.clientHeight / 2 + "px";
+					positions.tooltip.left =
+						position.left + triggerElement.clientWidth + 15 + "px";
+					positions.arrow.top = position.top + "px";
+					positions.arrow.left =
+						position.left + triggerElement.clientWidth + 10 + "px";
```

Comment:
I think the diff seems bigger than it is becaues the indentation is going from 2 spaces to 4 spaces. Do you think you could run the formatter to put the indentation back to 2 spaces?